toolkit/scripts/utils.mk

```
@@ -20,9 +20,25 @@ build_arch := $(shell uname -m)
 #
 # $1 - Folder path
 define create_folder
-$(shell if [ ! -d $1 ]; then mkdir -p $1 && touch -d @0 $1 ; fi )
+$(call shell_real_build_only, if [ ! -d $1 ]; then mkdir -p $1 && touch -d @0 $1 ; fi )
 endef
 
+# Runs a shell commannd only if we are actually doing a build rather than parsing the makefile for tab-completion etc
+# Make will automatically create the MAKEFLAGS variable which contains each of the flags, non-build commmands will include -n
+# which is the short form of --dry-run.
+#
+# $1 - The full command to run, if we are not doing --dry-run
+ifeq (n,$(findstring n,$(firstword $(MAKEFLAGS))))
+define shell_real_build_only
+endef
+else # ifeq (n,$(findstring...
+define shell_real_build_only
+$(shell $1)
+endef
```

Since it's so short, what do you think about this:
```suggestion
shell_real_build_only =
else # ifeq (n,$(findstring...
shell_real_build_only = $(shell $1)
endef
```

---

That was my first implementation as well, unfortunately Make doesn't parse it correctly. The ordering of the defines and if/else are important, although I'm not 100% sure what the exact issue is.

---

Not sure, if we're looking at the same thing. My suggestion is not about ordering but about replacing `define...endef` with a one-liner.  However, if it cannot easily be done in a single line, then there's no point to waste time on this.

---

oh, I see, got confused by the diff a bit.

I don't think there is a way to get around using the define operator, since `$(call` is what populates the `$1` arg and I want this to be a drop-in replacement for a call to `$(shell <arg>)`.